addons/portal_rating/models/mail_message.py

```diff
@@ -40,12 +40,19 @@ def _portal_message_format(self, properties_names, options=None):
             for rating in related_rating
         }
 
+        rating_stats_by_model = {
+            model: {
+                res_id: self.env[model].browse(res_id).sudo().rating_get_stats()
+                for res_id in {res_id for res_id in model_messages.mapped('res_id') if res_id}
```

Comment:
Better to browse all at once and iterate over records so prefetch_ids is set on the recordset and we avoid doing multiple fetch/read queries
```suggestion
                record_sudo.id: record_sudo.rating_get_stats()
                for record_sudo in self.env[model].browse({res_id for res_id in model_messages.mapped('res_id') if res_id}).sudo()
```